lib/TokenManager.php

```diff
@@ -147,6 +148,21 @@ public function getToken($fileId, $shareToken = null, $editoruid = null, $direct
 						}
 					}
 				}
+
+				// disable download if at least one shared access has it disabled
+				foreach ($files as $file) {
+					$storage = $file->getStorage();
+					// using string as we have no guarantee that "files_sharing" app is loaded
+					if ($storage->instanceOfStorage('OCA\Files_Sharing\SharedStorage')) {
```

Comment:
```suggestion
					if ($storage->instanceOfStorage(OCA\Files_Sharing\SharedStorage::class)) {
```

Should work too even if the app is disabled